src/settings.rs

```diff
@@ -272,6 +279,25 @@ impl Settings {
             .unwrap_or_else(|_| "<invalid db>".to_owned());
         format!("http://{}:{} ({}) {}", self.host, self.port, db, quota)
     }
+
+    pub fn build_cors(&self) -> Cors {
+        // Followed by the "official middleware" so they run first.
+        // actix is getting increasingly tighter about CORS headers. Our server is
+        // not a huge risk but does deliver XHR JSON content.
+        // For now, let's be permissive and use NGINX (the wrapping server)
+        // for finer grained specification.
+        let mut cors = Cors::permissive();
```

Comment:
it wasn't too happy with me when I used `Cors::default()`...  so I used permissive and below overrode allowed origin, if it was available?

Reply:
IIRC, `Cors::default()` is completely restrictive. You're supposed to add what you're using. `Cors::permissive()` is the opposite. For good security hygene we should probably be as explicit.

so:
```
cors = cors
	.allow_methods(vec![ "DELETE", "GET", "POST", "PUT"])
	.allow_headers(vec![
		"Authorization",
		"Content-Type",
		"UserAgent",
		crate::web::X_LAST_MODIFIED,
		crate::web::X_WEAVE_TIMESTAMP,
		crate::web::X_WEAVE_NEXT_OFFSET,
		crate::web::X_WEAVE_RECORDS,
		// might want to add these to `crate::web` so we're consistent.
		"X-Weave-Bytes",
		"X-Weave-Total-Records",
		"X-Weave-Total-Bytes",
		"X-Verify-Code",
		// I think this is only used for tests.
		"TEST_IDLES"
	]);
```

Definitely want to verify those. I just did a quick scan of `server/mod.rs` for the methods and a file scan for `headers().get(` to find the Header values. Obviously, before we push we'll want QA to run a client test to make sure things are working right. You could always fire up a local profile that points to a locally running server if you want to make their lives  bit easier.